home-assistant: 2020.12.2 -> 2021.1.0 (+entourage)

mweinelt opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

https://www.home-assistant.io/blog/2021/01/06/release-20211/

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

thanegill committed:
pythonPackage.uvicorn Fix tests on darwin

mweinelt committed:
python3Packages.zigpy: 0.26.0 -> 0.29.0

mweinelt committed:
python3Packages.zha-quirks: 0.0.47 -> 0.0.51

Fetch from GitHub as they removed tests from the PyPi tarball.

mweinelt committed:
python3Packages.zigpy-deconz: 0.11.0 -> 0.11.1

Fetch source from GitHub, the tests were removed from the PyPi tarball.
Also migrate to pytestCheckHook.

mweinelt committed:
python3Packages.zigpy-zigate: 0.7.3 -> 0.7.4

mweinelt committed:
python3Packages.h11: 0.9.0 -> 0.12.0

fabaff committed:
python3Packages.pyarlo: init at 0.2.4

mweinelt committed:
python3Packages.openwebifpy: 3.1.1 -> 3.2.7

mweinelt committed:
python3Packages.uvicorn: 0.11.5 -> 0.13.2

mweinelt commented:
I've prepared these updates during the home-assistant beta release cycle and therefore missed a few independent library updates, like:

- #106778
- #108308

This also supersedes
- #104006 (I'll pick up the darwin test fixes)

mweinelt committed:
home-assistant: 2020.12.2 -> 2021.1.0

mweinelt commented:
There is a slighty newer version of uvicorn (0.13.3) out there, but it creates circular dependencies with httpx during the testPhase.  I didn't want to deal with that in here.

mweinelt commented:
Result of `nixpkgs-review pr 108602` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.databases</li>
    <li>python39Packages.fastapi</li>
    <li>python39Packages.httpcore</li>
    <li>python39Packages.orm</li>
    <li>python39Packages.pyarlo</li>
    <li>python39Packages.starlette</li>
    <li>python39Packages.uvicorn</li>
  </ul>
</details>
<details>
  <summary>13 packages failed to build:</summary>
  <ul>
    <li>google-music-scripts</li>
    <li>python37Packages.entrance</li>
    <li>python37Packages.entrance-with-router-features</li>
    <li>python37Packages.google-music</li>
    <li>python37Packages.sanic-auth</li>
    <li>python37Packages.zigpy-znp</li>
    <li>python38Packages.entrance</li>
    <li>python38Packages.entrance-with-router-features</li>
    <li>python38Packages.google-music</li>
    <li>python38Packages.orm</li>
    <li>python38Packages.sanic-auth</li>
    <li>python38Packages.zigpy-znp</li>
    <li>python39Packages.zigpy-znp</li>
  </ul>
</details>
<details>
  <summary>69 packages built:</summary>
  <ul>
    <li>cura</li>
    <li>datasette (python38Packages.datasette)</li>
    <li>gns3-gui</li>
    <li>gns3-server</li>
    <li>home-assistant</li>
    <li>mirage-im</li>
    <li>mitmproxy (python38Packages.mitmproxy)</li>
    <li>octoprint</li>
    <li>pantalaimon</li>
    <li>python37Packages.PyRMVtransport</li>
    <li>python37Packages.aiosqlite</li>
    <li>python37Packages.alerta-server</li>
    <li>python37Packages.bellows</li>
    <li>python37Packages.databases</li>
    <li>python37Packages.datasette</li>
    <li>python37Packages.fastapi</li>
    <li>python37Packages.h11</li>
    <li>python37Packages.httpcore</li>
    <li>python37Packages.httpx</li>
    <li>python37Packages.matrix-nio</li>
    <li>python37Packages.mitmproxy</li>
    <li>python37Packages.openwebifpy</li>
    <li>python37Packages.orm</li>
    <li>python37Packages.pyarlo</li>
    <li>python37Packages.sanic</li>
    <li>python37Packages.sentry-sdk</li>
    <li>python37Packages.starlette</li>
    <li>python37Packages.uvicorn</li>
    <li>python37Packages.wsproto</li>
    <li>python37Packages.zha-quirks</li>
    <li>python37Packages.zigpy</li>
    <li>python37Packages.zigpy-deconz</li>
    <li>python38Packages.PyRMVtransport</li>
    <li>python38Packages.aiosqlite</li>
    <li>python38Packages.alerta-server</li>
    <li>python38Packages.bellows</li>
    <li>python38Packages.databases</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.h11</li>
    <li>python38Packages.httpcore</li>
    <li>python38Packages.httpx</li>
    <li>python38Packages.matrix-nio</li>
    <li>python38Packages.openwebifpy</li>
    <li>python38Packages.pyarlo</li>
    <li>python38Packages.sanic</li>
    <li>python38Packages.sentry-sdk</li>
    <li>python38Packages.starlette</li>
    <li>python38Packages.uvicorn</li>
    <li>python38Packages.wsproto</li>
    <li>python38Packages.zha-quirks</li>
    <li>python38Packages.zigpy</li>
    <li>python38Packages.zigpy-cc</li>
    <li>python38Packages.zigpy-deconz</li>
    <li>python38Packages.zigpy-xbee</li>
    <li>python38Packages.zigpy-zigate</li>
    <li>python39Packages.aiosqlite</li>
    <li>python39Packages.bellows</li>
    <li>python39Packages.h11</li>
    <li>python39Packages.matrix-nio</li>
    <li>python39Packages.mitmproxy</li>
    <li>python39Packages.openwebifpy</li>
    <li>python39Packages.wsproto</li>
    <li>python39Packages.zha-quirks</li>
    <li>python39Packages.zigpy</li>
    <li>python39Packages.zigpy-cc</li>
    <li>python39Packages.zigpy-deconz</li>
    <li>python39Packages.zigpy-xbee</li>
    <li>python39Packages.zigpy-zigate</li>
    <li>weechatScripts.weechat-matrix</li>
  </ul>
</details>

mweinelt commented:
Whoops, this got somewhat out of hand. Anyway, the remaining issues are
on the target branch and I think we should move this through quickly.

Except maybe the `zigpy-znp` test issues. They're timing out after 
latest`zigpy` update. I'm not yet sure why. CC @mvnetbiz

Result of `nixpkgs-review pr 108602` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.databases</li>
    <li>python39Packages.fastapi</li>
    <li>python39Packages.httpcore</li>
    <li>python39Packages.orm</li>
    <li>python39Packages.pyarlo</li>
    <li>python39Packages.starlette</li>
    <li>python39Packages.uvicorn</li>
  </ul>
</details>
<details>
  <summary>18 packages failed to build:</summary>
  <ul>
    <li>google-music-scripts</li>
    <li>python37Packages.audio-metadata</li>
    <li>python37Packages.google-music</li>
    <li>python37Packages.google-music-proto</li>
    <li>python37Packages.google-music-utils</li>
    <li>python37Packages.labelbox</li>
    <li>python37Packages.salmon-mail</li>
    <li>python37Packages.zigpy-znp</li>
    <li>python38Packages.audio-metadata</li>
    <li>python38Packages.google-music</li>
    <li>python38Packages.google-music-proto</li>
    <li>python38Packages.google-music-utils</li>
    <li>python38Packages.labelbox</li>
    <li>python38Packages.orm</li>
    <li>python38Packages.salmon-mail</li>
    <li>python38Packages.zigpy-znp</li>
    <li>python39Packages.zigpy-znp</li>
    <li>tts</li>
  </ul>
</details>
<details>
  <summary>121 packages built:</summary>
  <ul>
    <li>apache-airflow (python37Packages.apache-airflow)</li>
    <li>check_systemd</li>
    <li>cura</li>
    <li>datasette (python38Packages.datasette)</li>
    <li>evdevremapkeys</li>
    <li>gns3-gui</li>
    <li>gns3-server</li>
    <li>home-assistant</li>
    <li>keepkey_agent (python38Packages.keepkey_agent)</li>
    <li>ledger_agent (python38Packages.ledger_agent)</li>
    <li>luigi</li>
    <li>mirage-im</li>
    <li>mitmproxy (python38Packages.mitmproxy)</li>
    <li>mnemosyne</li>
    <li>octoprint</li>
    <li>pantalaimon</li>
    <li>python37Packages.PyRMVtransport</li>
    <li>python37Packages.aiosqlite</li>
    <li>python37Packages.alerta-server</li>
    <li>python37Packages.ansible-kernel</li>
    <li>python37Packages.ansible-runner</li>
    <li>python37Packages.bellows</li>
    <li>python37Packages.clldutils</li>
    <li>python37Packages.csvw</li>
    <li>python37Packages.databases</li>
    <li>python37Packages.datasette</li>
    <li>python37Packages.entrance</li>
    <li>python37Packages.entrance-with-router-features</li>
    <li>python37Packages.fastapi</li>
    <li>python37Packages.gtts</li>
    <li>python37Packages.h11</li>
    <li>python37Packages.httpcore</li>
    <li>python37Packages.httpx</li>
    <li>python37Packages.keepkey_agent</li>
    <li>python37Packages.ledger_agent</li>
    <li>python37Packages.libagent</li>
    <li>python37Packages.matrix-nio</li>
    <li>python37Packages.mitmproxy</li>
    <li>python37Packages.nagiosplugin</li>
    <li>python37Packages.ndjson</li>
    <li>python37Packages.openwebifpy</li>
    <li>python37Packages.orm</li>
    <li>python37Packages.phonemizer</li>
    <li>python37Packages.pyarlo</li>
    <li>python37Packages.python-daemon</li>
    <li>python37Packages.rfc3986</li>
    <li>python37Packages.sanic</li>
    <li>python37Packages.sanic-auth</li>
    <li>python37Packages.segments</li>
    <li>python37Packages.sentry-sdk</li>
    <li>python37Packages.starlette</li>
    <li>python37Packages.tbm-utils</li>
    <li>python37Packages.toggl-cli</li>
    <li>python37Packages.trezor_agent</li>
    <li>python37Packages.twine</li>
    <li>python37Packages.uvicorn</li>
    <li>python37Packages.wsproto</li>
    <li>python37Packages.zha-quirks</li>
    <li>python37Packages.zigpy</li>
    <li>python37Packages.zigpy-deconz</li>
    <li>python38Packages.PyRMVtransport</li>
    <li>python38Packages.aiosqlite</li>
    <li>python38Packages.alerta-server</li>
    <li>python38Packages.ansible-kernel</li>
    <li>python38Packages.ansible-runner</li>
    <li>python38Packages.bellows</li>
    <li>python38Packages.clldutils</li>
    <li>python38Packages.csvw</li>
    <li>python38Packages.databases</li>
    <li>python38Packages.entrance</li>
    <li>python38Packages.entrance-with-router-features</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.gtts</li>
    <li>python38Packages.h11</li>
    <li>python38Packages.httpcore</li>
    <li>python38Packages.httpx</li>
    <li>python38Packages.libagent</li>
    <li>python38Packages.matrix-nio</li>
    <li>python38Packages.nagiosplugin</li>
    <li>python38Packages.ndjson</li>
    <li>python38Packages.openwebifpy</li>
    <li>python38Packages.phonemizer</li>
    <li>python38Packages.pyarlo</li>
    <li>python38Packages.python-daemon</li>
    <li>python38Packages.rfc3986</li>
    <li>python38Packages.sanic</li>
    <li>python38Packages.sanic-auth</li>
    <li>python38Packages.segments</li>
    <li>python38Packages.sentry-sdk</li>
    <li>python38Packages.starlette</li>
    <li>python38Packages.tbm-utils</li>
    <li>python38Packages.toggl-cli</li>
    <li>trezor_agent (python38Packages.trezor_agent)</li>
    <li>python38Packages.twine</li>
    <li>python38Packages.uvicorn</li>
    <li>python38Packages.wsproto</li>
    <li>python38Packages.zha-quirks</li>
    <li>python38Packages.zigpy</li>
    <li>python38Packages.zigpy-cc</li>
    <li>python38Packages.zigpy-deconz</li>
    <li>python38Packages.zigpy-xbee</li>
    <li>python38Packages.zigpy-zigate</li>
    <li>python39Packages.aiosqlite</li>
    <li>python39Packages.bellows</li>
    <li>python39Packages.clldutils</li>
    <li>python39Packages.csvw</li>
    <li>python39Packages.h11</li>
    <li>python39Packages.matrix-nio</li>
    <li>python39Packages.mitmproxy</li>
    <li>python39Packages.openwebifpy</li>
    <li>python39Packages.phonemizer</li>
    <li>python39Packages.rfc3986</li>
    <li>python39Packages.segments</li>
    <li>python39Packages.wsproto</li>
    <li>python39Packages.zha-quirks</li>
    <li>python39Packages.zigpy</li>
    <li>python39Packages.zigpy-cc</li>
    <li>python39Packages.zigpy-deconz</li>
    <li>python39Packages.zigpy-xbee</li>
    <li>python39Packages.zigpy-zigate</li>
    <li>weechatScripts.weechat-matrix</li>
  </ul>
</details>

mweinelt commented:
@GrahamcOfBorg test home-assistant

mweinelt commented:
@GrahamcOfBorg test home-assistant

mweinelt commented:
- Disabled a flaky test `test_error_posted_as_event`
- Disabled stripping, its not useful on the 6600+ python files and takes >2 minutes on my 6C/12T desktop.

mweinelt commented:
@GrahamcOfBorg test home-assistant

mweinelt commented:
Can't repro the aarch64-linux failure on ofBorg, neither on my local aarch64-linux machine, nur on x86_64-linux. :disappointed:

mweinelt commented:
Something is wrong.

```
❯ ./result/bin/hass --version
2020.12.1
```

mweinelt committed:
python3Packages.voluptuous: 0.12.0 -> 0.12.1

mweinelt committed:
home-assistant: don't strip

Neither the home-assistant nor the frontend contain strippable binaries,
but the stripping process will still iterate over 6600+ files and notice
that they're not in a strippable format.

On my 6C/12T desktop CPU this takes slightly over two minutes.

mweinelt committed:
home-assistant: fix packagen name in update.sh

mweinelt commented:
@GrahamcOfBorg test home-assistant :pray:

mweinelt commented:
@GrahamcOfBorg test home-assistant

mweinelt commented:
WFM.

![20210107_00h47m00s_grim](https://user-images.githubusercontent.com/131599/103833704-f072f100-5081-11eb-8ddd-5c8bcf0f7bdb.png)

SuperSandro2000 commented:
> * Disabled stripping, its not useful on the 6600+ python files and takes >2 minutes on my 6C/12T desktop.

Please revert that. The terminal spam is fixed by #108081 which I am now merging.

mweinelt commented:
> > * Disabled stripping, its not useful on the 6600+ python files and takes >2 minutes on my 6C/12T desktop.
> 
> Please revert that. The terminal spam is fixed by #108081 which I am now merging.

It still iterates over the whole package, which is part of the issue. Not sure why we should keep stripping enabled.

SuperSandro2000 commented:
> It still iterates over the whole package, which is part of the issue. Not sure why we should keep stripping enabled.

Ah, true. If there are is no native code than this should be fine.

mvnetbiz commented:
@mweinelt https://github.com/zigpy/zigpy-znp/commit/628d4a4235e9375b12221a6ef416ae11e367e527
I can't figure out how to make pytest skip that test though.
I tried either of
```nix
pytestFlags = [
  "--deselect=tests/application/test_requests.py::test_force_remove"
  "-k 'not test_force_remove'"
];
```

mvnetbiz commented:
Oh it's pytestFlagsArray.
Can you add
```nix
pytestFlagsArray = [
    "--deselect=tests/application/test_requests.py::test_force_remove"
  ];
```
in zigpy-znp?

mweinelt committed:
python3Packages.zigpy-znp: disable test_force_remove test case

The zigpy-znp package didn't sync up with changes to the zigpy 0.29 in
time for the home-assistant 2021.1.0 release, so let's disable this test
until the next release.

Co-Authored-By: Matt Votava <mvnetbiz@gmail.com>

mweinelt commented:
Nice find, let's use `disabledTests` instead though! It's a much leaner API.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook

mweinelt commented:
Result of `nixpkgs-review pr 108602` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.databases</li>
    <li>python39Packages.fastapi</li>
    <li>python39Packages.httpcore</li>
    <li>python39Packages.orm</li>
    <li>python39Packages.pyarlo</li>
    <li>python39Packages.starlette</li>
    <li>python39Packages.uvicorn</li>
  </ul>
</details>
<details>
  <summary>13 packages failed to build:</summary>
  <ul>
    <li>google-music-scripts</li>
    <li>python37Packages.audio-metadata</li>
    <li>python37Packages.google-music</li>
    <li>python37Packages.google-music-proto</li>
    <li>python37Packages.google-music-utils</li>
    <li>python37Packages.labelbox</li>
    <li>python38Packages.audio-metadata</li>
    <li>python38Packages.google-music</li>
    <li>python38Packages.google-music-proto</li>
    <li>python38Packages.google-music-utils</li>
    <li>python38Packages.labelbox</li>
    <li>python38Packages.orm</li>
    <li>tts</li>
  </ul>
</details>
<details>
  <summary>144 packages built:</summary>
  <ul>
    <li>apache-airflow (python37Packages.apache-airflow)</li>
    <li>appdaemon</li>
    <li>check_systemd</li>
    <li>cura</li>
    <li>datasette (python38Packages.datasette)</li>
    <li>elasticsearch-curator</li>
    <li>esphome</li>
    <li>evdevremapkeys</li>
    <li>gns3-gui</li>
    <li>gns3-server</li>
    <li>home-assistant</li>
    <li>keepkey_agent (python38Packages.keepkey_agent)</li>
    <li>ledger_agent (python38Packages.ledger_agent)</li>
    <li>luigi</li>
    <li>mirage-im</li>
    <li>mitmproxy (python38Packages.mitmproxy)</li>
    <li>mnemosyne</li>
    <li>octoprint</li>
    <li>pantalaimon</li>
    <li>python37Packages.PyRMVtransport</li>
    <li>python37Packages.aiosqlite</li>
    <li>python37Packages.alerta-server</li>
    <li>python37Packages.ansible-kernel</li>
    <li>python37Packages.ansible-runner</li>
    <li>python37Packages.async-upnp-client</li>
    <li>python37Packages.bellows</li>
    <li>python37Packages.clldutils</li>
    <li>python37Packages.csvw</li>
    <li>python37Packages.databases</li>
    <li>python37Packages.datasette</li>
    <li>python37Packages.entrance</li>
    <li>python37Packages.entrance-with-router-features</li>
    <li>python37Packages.fastapi</li>
    <li>python37Packages.gtts</li>
    <li>python37Packages.h11</li>
    <li>python37Packages.httpcore</li>
    <li>python37Packages.httpx</li>
    <li>python37Packages.keepkey_agent</li>
    <li>python37Packages.ledger_agent</li>
    <li>python37Packages.libagent</li>
    <li>python37Packages.matrix-nio</li>
    <li>python37Packages.mitmproxy</li>
    <li>python37Packages.nagiosplugin</li>
    <li>python37Packages.ndjson</li>
    <li>python37Packages.nestedtext</li>
    <li>python37Packages.openwebifpy</li>
    <li>python37Packages.orm</li>
    <li>python37Packages.phonemizer</li>
    <li>python37Packages.pyarlo</li>
    <li>python37Packages.pyhs100</li>
    <li>python37Packages.python-daemon</li>
    <li>python37Packages.rfc3986</li>
    <li>python37Packages.salmon-mail</li>
    <li>python37Packages.sanic</li>
    <li>python37Packages.sanic-auth</li>
    <li>python37Packages.segments</li>
    <li>python37Packages.sentry-sdk</li>
    <li>python37Packages.starlette</li>
    <li>python37Packages.tbm-utils</li>
    <li>python37Packages.toggl-cli</li>
    <li>python37Packages.trezor_agent</li>
    <li>python37Packages.twine</li>
    <li>python37Packages.uvicorn</li>
    <li>python37Packages.voluptuous</li>
    <li>python37Packages.voluptuous-serialize</li>
    <li>python37Packages.wsproto</li>
    <li>python37Packages.zha-quirks</li>
    <li>python37Packages.zigpy</li>
    <li>python37Packages.zigpy-deconz</li>
    <li>python37Packages.zigpy-znp</li>
    <li>python38Packages.PyRMVtransport</li>
    <li>python38Packages.aiosqlite</li>
    <li>python38Packages.alerta-server</li>
    <li>python38Packages.ansible-kernel</li>
    <li>python38Packages.ansible-runner</li>
    <li>python38Packages.async-upnp-client</li>
    <li>python38Packages.bellows</li>
    <li>python38Packages.clldutils</li>
    <li>python38Packages.csvw</li>
    <li>python38Packages.databases</li>
    <li>python38Packages.entrance</li>
    <li>python38Packages.entrance-with-router-features</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.gtts</li>
    <li>python38Packages.h11</li>
    <li>python38Packages.httpcore</li>
    <li>python38Packages.httpx</li>
    <li>python38Packages.libagent</li>
    <li>python38Packages.matrix-nio</li>
    <li>python38Packages.nagiosplugin</li>
    <li>python38Packages.ndjson</li>
    <li>python38Packages.nestedtext</li>
    <li>python38Packages.openwebifpy</li>
    <li>python38Packages.phonemizer</li>
    <li>python38Packages.pyarlo</li>
    <li>python38Packages.pyhs100</li>
    <li>python38Packages.python-daemon</li>
    <li>python38Packages.rfc3986</li>
    <li>python38Packages.salmon-mail</li>
    <li>python38Packages.sanic</li>
    <li>python38Packages.sanic-auth</li>
    <li>python38Packages.segments</li>
    <li>python38Packages.sentry-sdk</li>
    <li>python38Packages.starlette</li>
    <li>python38Packages.tbm-utils</li>
    <li>python38Packages.toggl-cli</li>
    <li>trezor_agent (python38Packages.trezor_agent)</li>
    <li>python38Packages.twine</li>
    <li>python38Packages.uvicorn</li>
    <li>python38Packages.voluptuous</li>
    <li>python38Packages.voluptuous-serialize</li>
    <li>python38Packages.wsproto</li>
    <li>python38Packages.zha-quirks</li>
    <li>python38Packages.zigpy</li>
    <li>python38Packages.zigpy-cc</li>
    <li>python38Packages.zigpy-deconz</li>
    <li>python38Packages.zigpy-xbee</li>
    <li>python38Packages.zigpy-zigate</li>
    <li>python38Packages.zigpy-znp</li>
    <li>python39Packages.aiosqlite</li>
    <li>python39Packages.async-upnp-client</li>
    <li>python39Packages.bellows</li>
    <li>python39Packages.clldutils</li>
    <li>python39Packages.csvw</li>
    <li>python39Packages.h11</li>
    <li>python39Packages.matrix-nio</li>
    <li>python39Packages.mitmproxy</li>
    <li>python39Packages.nestedtext</li>
    <li>python39Packages.openwebifpy</li>
    <li>python39Packages.phonemizer</li>
    <li>python39Packages.pyhs100</li>
    <li>python39Packages.rfc3986</li>
    <li>python39Packages.segments</li>
    <li>python39Packages.voluptuous</li>
    <li>python39Packages.voluptuous-serialize</li>
    <li>python39Packages.wsproto</li>
    <li>python39Packages.zha-quirks</li>
    <li>python39Packages.zigpy</li>
    <li>python39Packages.zigpy-cc</li>
    <li>python39Packages.zigpy-deconz</li>
    <li>python39Packages.zigpy-xbee</li>
    <li>python39Packages.zigpy-zigate</li>
    <li>python39Packages.zigpy-znp</li>
    <li>weechatScripts.weechat-matrix</li>
  </ul>
</details>

Mic92 commented:
Some of the broken python packages are not compatible with python3.9 yet. Like pyarlo

mweinelt commented:
Yep, I think there is nothing left to do here for me.

mvnetbiz commented:
builder for '/nix/store/ydn05m1q9i58jgzk950fk6m3z5fhpkv7-homeassistant-2021.1.0.drv' failed with exit code 1; last 10 log lines:
  tests/components/frontend/test_init.py::test_auth_authorize[pyloop]
  tests/components/frontend/test_init.py::test_static_paths[pyloop]
    /build/source/homeassistant/components/http/ban.py:48: DeprecationWarning: Changing state of started or joined application is deprecated
      app[KEY_BANNED_IPS] = await async_load_ip_bans_config(
  
  -- Docs: https://docs.pytest.org/en/stable/warnings.html
  =========================== short test summary info ============================
  FAILED tests/util/yaml/test_init.py::TestSecrets::test_secrets_credstash - ke...
  FAILED tests/util/yaml/test_init.py::TestSecrets::test_secrets_from_unrelated_fails
  ====== 2 failed, 2481 passed, 1 skipped, 10 warnings in 72.51s (0:01:12) =======

mvnetbiz commented on pkgs/servers/home-assistant/default.nix:
```diff
@@ -176,6 +179,8 @@ in with py.pkgs; buildPythonApplication rec {
     # ValueError: count must be a positive integer (got 0)
     "test_media_view"
+    # AssertionError: len(events) == 1
+    "test_error_posted_as_event"
```
```suggestion
    "test_error_posted_as_event"
    # No keyring backends are available
    "test_secrets_from_unrelated_fails"
    "test_secrets_credstash"
```

mweinelt replied:
Yeah, that sounds sensible. Will do!

There are so many tests that are failing and that I can't repro on my dev machines, it's a bit sad.

mweinelt replied:
So, the credstash one looks to be going away. I'm not so sure about `test_secrets_from_unrelated_fails`. Can you post the full traceback, please?

mvnetbiz replied:
```
homeassistant> ________________ TestSecrets.test_secrets_from_unrelated_fails _________________
homeassistant> [gw13] linux -- Python 3.8.6 /nix/store/papbnjfxwrdkmz6xfp7v2q7nxyvgcr5l-python3-3.8.6/bin/python3.8
homeassistant> self = <tests.util.yaml.test_init.TestSecrets testMethod=test_secrets_from_unrelated_fails>
homeassistant>     def test_secrets_from_unrelated_fails(self):
homeassistant>         """Test loading secrets from unrelated folder fails."""
homeassistant>         load_yaml(os.path.join(self._unrelated_path, yaml.SECRET_YAML), "test: failure")
homeassistant>         with pytest.raises(HomeAssistantError):
homeassistant> >           load_yaml(
homeassistant>                 os.path.join(self._sub_folder_path, "sub.yaml"),
homeassistant>                 "http:\n  api_password: !secret test",
homeassistant>             )
homeassistant> tests/util/yaml/test_init.py:393:
homeassistant> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
homeassistant> tests/util/yaml/test_init.py:294: in load_yaml
homeassistant>     return load_yaml_config_file(fname)
homeassistant> homeassistant/config.py:329: in load_yaml_config_file
homeassistant>     conf_dict = load_yaml(config_path)
homeassistant> homeassistant/util/yaml/loader.py:62: in load_yaml
homeassistant>     return parse_yaml(conf_file)
homeassistant> homeassistant/util/yaml/loader.py:73: in parse_yaml
homeassistant>     return yaml.load(content, Loader=SafeLineLoader) or OrderedDict()
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/__init__.py:114: in load
homeassistant>     return loader.get_single_data()
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:51: in get_single_data
homeassistant>     return self.construct_document(node)
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:55: in construct_document
homeassistant>     data = self.construct_object(node)
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:100: in construct_object
homeassistant>     data = constructor(self, node)
homeassistant> homeassistant/util/yaml/loader.py:201: in _ordered_dict
homeassistant>     nodes = loader.construct_pairs(node)
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:155: in construct_pairs
homeassistant>     value = self.construct_object(value_node, deep=deep)
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:100: in construct_object
homeassistant>     data = constructor(self, node)
homeassistant> homeassistant/util/yaml/loader.py:201: in _ordered_dict
homeassistant>     nodes = loader.construct_pairs(node)
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:155: in construct_pairs
homeassistant>     value = self.construct_object(value_node, deep=deep)
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:100: in construct_object
homeassistant>     data = constructor(self, node)
homeassistant> homeassistant/util/yaml/loader.py:299: in secret_yaml
homeassistant>     pwd = keyring.get_password(_SECRET_NAMESPACE, node.value)
homeassistant> /nix/store/015wnml2z5y083ygx0dz4hfq9k8ls2c2-python3.8-keyring-21.5.0/lib/python3.8/site-packages/keyring/core.py:53: in get_password
homeassistant>     return _keyring_backend.get_password(service_name, username)
homeassistant> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
homeassistant> self = <keyring.backends.fail.Keyring object at 0x7ffff44624c0>
homeassistant> service = 'homeassistant', username = 'test', password = None
homeassistant>     def get_password(self, service, username, password=None):
homeassistant>         msg = (
homeassistant>             "No recommended backend was available. Install a recommended 3rd "
homeassistant>             "party backend package; or, install the keyrings.alt package if "
homeassistant>             "you want to use the non-recommended backends. See "
homeassistant>             "https://pypi.org/project/keyring for details."
homeassistant>         )
homeassistant> >       raise NoKeyringError(msg)
homeassistant> E       keyring.errors.NoKeyringError: No recommended backend was available. Install a recommended 3rd party backend package; or, install the keyrings.alt package if you want to use the non-recommended backends. See https://pypi.org/project/keyring for details.
homeassistant> /nix/store/015wnml2z5y083ygx0dz4hfq9k8ls2c2-python3.8-keyring-21.5.0/lib/python3.8/site-packages/keyring/backends/fail.py:25: NoKeyringError
homeassistant> ----------------------------- Captured stderr call -----------------------------
homeassistant> DEBUG:tests.common:patch_yaml_files match /build/source/tests/testing_config/secrets.yaml
homeassistant> DEBUG:tests.common:patch_yaml_files match /build/source/tests/testing_config/configuration.yaml
homeassistant> DEBUG:homeassistant.util.yaml.loader:Loading /build/source/tests/testing_config/secrets.yaml
homeassistant> DEBUG:tests.common:patch_yaml_files match /build/source/tests/testing_config/secrets.yaml
homeassistant> DEBUG:homeassistant.util.yaml.loader:Secret http_pw retrieved from secrets.yaml in folder /build/source/tests/testing_config
homeassistant> DEBUG:homeassistant.util.yaml.loader:Secret comp1_un retrieved from secrets.yaml in folder /build/source/tests/testing_config
homeassistant> DEBUG:homeassistant.util.yaml.loader:Secret comp1_pw retrieved from secrets.yaml in folder /build/source/tests/testing_config
homeassistant> DEBUG:tests.common:patch_yaml_files match /build/source/tests/testing_config/unrelated/secrets.yaml
homeassistant> DEBUG:tests.common:patch_yaml_files match /build/source/tests/testing_config/subFolder/sub.yaml
homeassistant> DEBUG:homeassistant.util.yaml.loader:Loading /build/source/tests/testing_config/subFolder/secrets.yaml
homeassistant> DEBUG:homeassistant.util.yaml.loader:Loading /build/source/tests/secrets.yaml
homeassistant> DEBUG:homeassistant.util.yaml.loader:Loading /build/source/secrets.yaml
homeassistant> DEBUG:homeassistant.util.yaml.loader:Loading /build/secrets.yaml
homeassistant> ------------------------------ Captured log call -------------------------------
homeassistant> DEBUG    tests.common:common.py:714 patch_yaml_files match /build/source/tests/testing_config/secrets.yaml
homeassistant> DEBUG    tests.common:common.py:714 patch_yaml_files match /build/source/tests/testing_config/configuration.yaml
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:255 Loading /build/source/tests/testing_config/secrets.yaml
homeassistant> DEBUG    tests.common:common.py:714 patch_yaml_files match /build/source/tests/testing_config/secrets.yaml
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:283 Secret http_pw retrieved from secrets.yaml in folder /build/source/tests/testing_config
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:283 Secret comp1_un retrieved from secrets.yaml in folder /build/source/tests/testing_config
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:283 Secret comp1_pw retrieved from secrets.yaml in folder /build/source/tests/testing_config
homeassistant> DEBUG    tests.common:common.py:714 patch_yaml_files match /build/source/tests/testing_config/unrelated/secrets.yaml
homeassistant> DEBUG    tests.common:common.py:714 patch_yaml_files match /build/source/tests/testing_config/subFolder/sub.yaml
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:255 Loading /build/source/tests/testing_config/subFolder/secrets.yaml
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:255 Loading /build/source/tests/secrets.yaml
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:255 Loading /build/source/secrets.yaml
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:255 Loading /build/secrets.yaml
```

mweinelt replied:
On what target are you building this? I really wonder why I don't get any of these NoKeyringError exceptions. Are you on darwin?

mvnetbiz reviewed (changes requested):
I think one of the other components I have enabled has python keyring as a transitive dependency, so home-assistant runs these tests, but I do not use this feature, and there are no keyring backends installed so the test does not pass. I think we should just disable these tests, the feature is being deprecated in a few months. https://www.home-assistant.io/docs/tools/keyring/.

mweinelt commented:
On what target are you building this? I really wonder why I don't get any of these `NoKeyringError` exceptions. Are you on darwin?

mvnetbiz commented:
I'm on NixOS. Do you have the Google text to speech component enabled?

mweinelt commented:
> I'm on NixOS. Do you have the Google text to speech component enabled?

I have not, but it shouldn't really matter, as those are the build time tests.

mvnetbiz commented:
```term
$ nix why-depends --derivation /nix/store/krcyfhayl0b9r8ilvhz55bg9wc8g3w9a-unit-home-assistant.service.drv /nix/store/iysbhf7fh7jhdpi34qyacck5x79zd34f-python3.8-keyring-21.5.0.drv
/nix/store/krcyfhayl0b9r8ilvhz55bg9wc8g3w9a-unit-home-assistant.service.drv
└───/: …-4.8.drv",["out"]),("/nix/store/ydn05m1q9i58jgzk950fk6m3z5fhpkv7-homeassistant-2021.1.0.drv",["o…
    → /nix/store/ydn05m1q9i58jgzk950fk6m3z5fhpkv7-homeassistant-2021.1.0.drv
    └───/: ….7.0.drv",["out"]),("/nix/store/jx62hf0rlplq2gp9ksri3xnsdlapvqff-python3.8-gtts-2.2.1.drv",["out…
        → /nix/store/jx62hf0rlplq2gp9ksri3xnsdlapvqff-python3.8-gtts-2.2.1.drv
        └───/: ….1.4.drv",["out"]),("/nix/store/pv0drdv9n60y8l6qkqlskmvydac31xmi-python3.8-twine-3.2.0.drv",["ou…
            → /nix/store/pv0drdv9n60y8l6qkqlskmvydac31xmi-python3.8-twine-3.2.0.drv
            └───/: ….6.1.drv",["out"]),("/nix/store/iysbhf7fh7jhdpi34qyacck5x79zd34f-python3.8-keyring-21.5.0.drv",[…
                → /nix/store/iysbhf7fh7jhdpi34qyacck5x79zd34f-python3.8-keyring-21.5.0.drv
```

mvnetbiz commented:
I think the test only runs if keyring is installed, you probably don't have it.

mweinelt commented:
Yeah, no keyring here, I don't think my user env would influence a sandboxed package build though, very weird. 

Pushed the updated package, please retest.

mvnetbiz commented:
Oh the error above for me is for my own H-A config, that doesn't come from the nixos test config, sorry if that is off topic?

mweinelt commented:
Oh, okay. Certainly isn't!

mvnetbiz commented:
Everything appears to be working for me, zigbee included.

SuperSandro2000 commented:
> Oh it's pytestFlagsArray.

and in the next merge of staging we also have disableTestFiles.

etu reviewed (approved):
I haven't reviewed the code, but I have built this branch on my home assistant at home and it works with both Z-Wave and Zigbee for me which have been blocking that box for updates for a while now. :+1:

SuperSandro2000 commented:
@mweinelt Sorry about the merge conflicts. I touched some of those python packages.

wd15 commented:
Result of `nixpkgs-review pr 108602` [1](https://github.com/Mic92/nixpkgs-review                            
)                                                   
<details>                                                                                                   
  <summary>3 package marked as broken and skipped:</summary>                                                
                                                      
  - python39Packages.httpcore                         
  - python39Packages.pyarlo                                                                                 
  - python39Packages.uvicorn                        
</details>                                            
<details>                                             
  <summary>1 package failed to build:</summary>                                                          
                                                      
  - google-music-scripts                              
</details>                                                                                                  
<details>                                           
  <summary>81 package built:</summary>              

  - appdaemon                                       
  - datasette (python38Packages.datasette)                                                               
  - elasticsearch-curator                           
  - esphome                                         
  - home-assistant                                  
  - mirage-im                                       
  - mitmproxy (python38Packages.mitmproxy)                                                               
  - pantalaimon                                     
  - python37Packages.PyRMVtransport                 
  - python37Packages.async-upnp-client              
  - python37Packages.bellows                        
  - python37Packages.datasette                      
  - python37Packages.entrance                       
  - python37Packages.entrance-with-router-features                                                       
  - python37Packages.google-music                   
  - python37Packages.h11                            
  - python37Packages.httpcore                       
  - python37Packages.httpx                          
  - python37Packages.matrix-nio                     
  - python37Packages.mitmproxy                      
  - python37Packages.nestedtext                     
  - python37Packages.openwebifpy                    
  - python37Packages.pyarlo                         
  - python37Packages.pyhs100                        
  - python37Packages.sanic                          
  - python37Packages.sanic-auth                     
  - python37Packages.uvicorn                        
  - python37Packages.voluptuous                     
  - python37Packages.voluptuous-serialize                                                                
  - python37Packages.wsproto                        
  - python37Packages.zha-quirks                     
  - python37Packages.zigpy             
  - python38Packages.zigpy-cc
  - python38Packages.zigpy-deconz
  - python38Packages.zigpy-xbee
  - python38Packages.zigpy-zigate
  - python38Packages.zigpy-znp
  - python39Packages.async-upnp-client
  - python39Packages.bellows
  - python39Packages.h11
  - python39Packages.matrix-nio
  - python39Packages.mitmproxy
  - python39Packages.nestedtext
  - python39Packages.openwebifpy
  - python39Packages.pyhs100
  - python39Packages.voluptuous
  - python39Packages.voluptuous-serialize
  - python39Packages.wsproto
  - python39Packages.zha-quirks
  - python39Packages.zigpy
  - python39Packages.zigpy-cc
  - python39Packages.zigpy-deconz
  - python39Packages.zigpy-xbee
  - python39Packages.zigpy-zigate
  - python39Packages.zigpy-znp
  - weechatScripts.weechat-matrix
</details>

pinpox commented:
Some tests are still failing for me. Running `nix-shell -p pkgs.home-assistant --run "hass --version"` gives me this error:

```

...

Requirement already satisfied: multidict<7.0,>=4.5 in /nix/store/jmrq36f6zw334nww7zk532703r43daja-python3.8-multidict-5.0.2/lib/python3.8/site-packages (from aiohttp==3.7.3->homeassistant==2021.1.3) (5.0.2)
Installing collected packages: homeassistant
Successfully installed homeassistant-2021.1.3
/build/source
Finished executing pipInstallPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/q1qavlaq21m2ihdz8hnxx6f43cg98f01-homeassistant-2021.1.3
patching script interpreter paths in /nix/store/q1qavlaq21m2ihdz8hnxx6f43cg98f01-homeassistant-2021.1.3
checking for references to /build/ in /nix/store/q1qavlaq21m2ihdz8hnxx6f43cg98f01-homeassistant-2021.1.3...
Rewriting #!/nix/store/yi9a1fafijn0nf5m8qkmnl955vi2mncm-python3-3.8.6/bin/python3.8 to #!/nix/store/yi9a1fafijn0nf5m8qkmnl955vi2mncm-python3-3.8.6
wrapping `/nix/store/q1qavlaq21m2ihdz8hnxx6f43cg98f01-homeassistant-2021.1.3/bin/hass'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
no Makefile or custom installCheckPhase, doing nothing
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.8.6, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /build/source, configfile: pyproject.toml
plugins: aiohttp-0.3.0, xdist-2.1.0, forked-1.3.0, requests-mock-1.8.0
gw0 [2503] / gw1 [2503] / gw2 [2503] / gw3 [2503] / gw4 [2503] / gw5 [2503]
........................................................................ [  2%]
......................................................................... [  5%]
......................................................................... [  8%]
........................................................................ [ 11%]
......................................................................... [ 14%]
......................................................................... [ 17%]
........................................................................ [ 20%]
........................................................................ [ 23%]
.......................................................................... [ 25%]
........................................................................ [ 28%]
........................................................................ [ 31%]
........................................................................ [ 34%]
......................................................................... [ 37%]
......................................................................... [ 40%]
.......................................................................... [ 43%]
........................................................................ [ 46%]
........................................................................ [ 49%]
........................................................................ [ 52%]
........................................................................ [ 54%]
......................................................................... [ 57%]
......................................................................... [ 60%]
........................................................................ [ 63%]
........................................................................ [ 66%]
............................................................................ [ 69%]
........................................................................ [ 72%]
.......................................................................... [ 75%]
........................................................................ [ 78%]
........................................................................ [ 81%]
.......................................................................... [ 84%]
.......................................................................... [ 86%]
......................................................................... [ 89%]
........................................................................ [ 92%]
......................s................................................. [ 95%]
.......................................................................... [ 98%]
..............................                                           [100%]
=============================== warnings summary ===============================
tests/helpers/test_script.py::test_condition_created_once[pyloop]
  /nix/store/gnyhnxssmpw6qa16jx3dyan45ldrpikf-python3.8-pytest-6.1.2/lib/python3.8/site-packages/_pytest/store.py:121: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    del self._store[key]

tests/components/frontend/test_init.py::test_get_panels[pyloop]
tests/components/frontend/test_init.py::test_auth_load[pyloop]
tests/components/frontend/test_init.py::test_onboarding_load[pyloop]
tests/components/frontend/test_init.py::test_auth_authorize[pyloop]
tests/components/frontend/test_init.py::test_static_paths[pyloop]
tests/components/frontend/test_init.py::test_frontend_and_static[pyloop]
tests/components/frontend/test_init.py::test_dont_cache_service_worker[pyloop]
tests/components/frontend/test_init.py::test_404[pyloop]
tests/components/frontend/test_init.py::test_we_cannot_POST_to_root[pyloop]
  /build/source/homeassistant/components/http/ban.py:48: DeprecationWarning: Changing state of started or joined application is deprecated
    app[KEY_BANNED_IPS] = await async_load_ip_bans_config(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========== 2502 passed, 1 skipped, 10 warnings in 67.23s (0:01:07) ============
Finished executing pytestCheckPhase
pytestcachePhase
2021.1.3

```

mweinelt commented:
Yes?

pinpox commented:
@mweinelt sorry, I accidentaly deleted my comment. `home-manager` is failing to build because of some tests still:
```
▸ nix-build ./krops.nix -A birne && ./result
/nix/store/1qaii67qh7r8hdkdmqgw2y762bx8f55b-deploy-birne
+ /nix/store/grm0kmhcm8zld7i6l2lfiw6c1s504qjp-populate.192.168.2.84.machine-config
sending incremental file list
./
tags
.git/FETCH_HEAD
.git/objects/
modules/home-assistant/
modules/home-assistant/default.nix

sent 65,352 bytes  received 431 bytes  131,566.00 bytes/sec
total size is 9,751,245  speedup is 148.23
+ /nix/store/a0zv3w9sjilgf73icpnfv4d8fwkn76i8-populate.192.168.2.84.secrets
sending incremental file list
./
borg-server/
borg/
cloudflare/
monit/
ssh/
ssh/root/
wireguard/

sent 468 bytes  received 47 bytes  343.33 bytes/sec
total size is 1,212  speedup is 2.35
warning: Git tree '/var/src/machine-config' is dirty
building the system configuration...
warning: Git tree '/var/src/machine-config' is dirty
building '/nix/store/dabjnd0hc56zc8aas8305bcsq7zl50m9-homeassistant-2021.1.3.drv'...
building '/nix/store/54g8lfs08z8nb1946jg22xm60m5jrdm6-configuration.json.drv'...
building '/nix/store/skyfjw629ydnc17qy105w08axwm26qfn-configuration.yaml.drv'...
building '/nix/store/zmd7ln1avpm0sj048zkgipjivfacnlaw-unit-script-home-assistant-pre-start.drv'...
error: --- Error --- nix
builder for '/nix/store/dabjnd0hc56zc8aas8305bcsq7zl50m9-homeassistant-2021.1.3.drv' failed with exit code 1; last 10 log lines:
      app[KEY_BANNED_IPS] = await async_load_ip_bans_config(

  tests/components/shell_command/test_init.py::test_do_no_run_forever[pyloop]
    /nix/store/yi9a1fafijn0nf5m8qkmnl955vi2mncm-python3-3.8.6/lib/python3.8/asyncio/unix_events.py:917: RuntimeWarning: A loop is being detached from a child watcher with pending handlers
      warnings.warn(

  -- Docs: https://docs.pytest.org/en/stable/warnings.html
  =========================== short test summary info ============================
  ERROR tests/components/ssdp/test_init.py::test_scan_match_st[pyloop] - Runtim...
  ====== 2501 passed, 1 skipped, 11 warnings, 1 error in 169.07s (0:02:49) =======
error: --- Error --- nix
1 dependencies of derivation '/nix/store/7mbz7wqckfa19ls5milsmdfmxxj7ff07-unit-home-assistant.service.drv' failed to build
error: --- Error --- nix
1 dependencies of derivation '/nix/store/0ivfl11fc93vnz4bka2alif5n53pwfnb-system-units.drv' failed to build
error: --- Error --- nix
1 dependencies of derivation '/nix/store/gnc9yspvc6crf8n7f9irdgbqcdcxv4m6-etc.drv' failed to build
error: --- Error --- nix
1 dependencies of derivation '/nix/store/0ss54wzr3m8ycmh0sl3bxsbibjanx3fd-nixos-system-birne-21.03.20210117.68398d2.drv' failed to build
warning: Git tree '/var/src/machine-config' is dirty
building the system configuration...
warning: Git tree '/var/src/machine-config' is dirty
building '/nix/store/dabjnd0hc56zc8aas8305bcsq7zl50m9-homeassistant-2021.1.3.drv'...
error: --- Error --- nix
builder for '/nix/store/dabjnd0hc56zc8aas8305bcsq7zl50m9-homeassistant-2021.1.3.drv' failed with exit code 1; last 10 log lines:
      app[KEY_BANNED_IPS] = await async_load_ip_bans_config(

  tests/components/shell_command/test_init.py::test_do_no_run_forever[pyloop]
    /nix/store/yi9a1fafijn0nf5m8qkmnl955vi2mncm-python3-3.8.6/lib/python3.8/asyncio/unix_events.py:917: RuntimeWarning: A loop is being detached from a child watcher with pending handlers
      warnings.warn(

  -- Docs: https://docs.pytest.org/en/stable/warnings.html
  =========================== short test summary info ============================
  ERROR tests/components/ssdp/test_init.py::test_scan_match_st[pyloop] - Runtim...
  ====== 2501 passed, 1 skipped, 11 warnings, 1 error in 168.88s (0:02:48) =======
error: --- Error --- nix
1 dependencies of derivation '/nix/store/7mbz7wqckfa19ls5milsmdfmxxj7ff07-unit-home-assistant.service.drv' failed to build
error: --- Error --- nix
1 dependencies of derivation '/nix/store/0ivfl11fc93vnz4bka2alif5n53pwfnb-system-units.drv' failed to build
error: --- Error --- nix
1 dependencies of derivation '/nix/store/gnc9yspvc6crf8n7f9irdgbqcdcxv4m6-etc.drv' failed to build
error: --- Error --- nix
1 dependencies of derivation '/nix/store/0ss54wzr3m8ycmh0sl3bxsbibjanx3fd-nixos-system-birne-21.03.20210117.68398d2.drv' failed to build
```

I'm tryin to use it with this minimal config: https://gist.github.com/pinpox/9a6b5bd1d0f7cee3827f1106e35eeebd

mweinelt commented:
Can you please post the complete test phase output, where things go wrong? I have a feeling what kind of issue this is, but I want to confirm. Generally some tests experience flakyness/racyness, sadly.

pinpox commented:
@mweinelt How can I get more output? The deployment command above is mostly just running `nixos-rebuild switch -v --show-trace --flake ${targetPath}/machine-config` on the server, not sure how to get the testPhase output

mweinelt commented:
Please ask the maintainers of your deployment tool (krops), I'm not familiar with it. Also please note that home-assistant is a pretty big package, so 2-3 minutes to build and test it are expected.

pinpox commented:
@mweinelt I'm not sure why, but re-running the rebuild just succeeded. I'm still getting a 404 on the page though, but that might be a different issue with my config. Would it be possible to disable the python tests for the package somehow?

Mic92 commented:
@pinbox Usually you can do `nix log /nix/store/dabjnd0hc56zc8aas8305bcsq7zl50m9-homeassistant-2021.1.3.drv` to get the full log after having a failed build. And nix-collect-garbage to remove a succeeding build.

mweinelt commented:
You can disable the tests by overriding `doCheck` like so:

```
  services.home-assistant.package = (pkgs.home-assistant.overrideAttrs (old: {
    doCheck = false;
  });
```

pinpox commented:
@Mic92 I ran `sudo nix-collect-garbage` then build on the machine itself (without krops) again successfully:

```
pinpox@birne /var/src/machine-config main* 7s
▸ sudo nixos-rebuild -v --show-trace --flake ".#birne" switch
warning: Git tree '/var/src/machine-config' is dirty
building the system configuration...
warning: Git tree '/var/src/machine-config' is dirty
updating GRUB 2 menu...
activating the configuration...
setting up /etc...
reloading user units for pinpox...
setting up tmpfiles
the following new units were started: cfdyndns.service
```
I can see, that home-assistant is running:

```

pinpox@birne /var/src/machine-config main*
▸ netstat -tulpn
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      -
tcp        0      0 127.0.0.1:8123          0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      -
tcp        0      0 192.168.7.4:2812        0.0.0.0:*               LISTEN      -
tcp6       0      0 :::80                   :::*                    LISTEN      -
tcp6       0      0 :::22                   :::*                    LISTEN      -
tcp6       0      0 :::443                  :::*                    LISTEN      -
udp        0      0 0.0.0.0:68              0.0.0.0:*                           -
udp        0      0 0.0.0.0:58460           0.0.0.0:*                           -
udp6       0      0 :::546                  :::*                                -
udp6       0      0 :::58460                :::*                                -
```

But I can't access it. I tried using the configured url or the localhost IP on the machine itself:
```
pinpox@birne /var/src/machine-config main*
▸ curl 127.0.0.1:8123
404: Not Found
```

The configuration is this: https://termbin.com/j5d3

mweinelt commented:
You are missing important modules, that are part of `default_config` (https://www.home-assistant.io/integrations/default_config/).

Enable it by setting 

```
services.home-assistant.config.default_config = {};
```

pinpox commented:
@mweinelt I'm back to failing. Here is more output regarding the python tests:

```

▸ sudo nixos-rebuild -v --show-trace --flake ".#birne" switch
[sudo] password for pinpox:
warning: Git tree '/var/src/machine-config' is dirty
building the system configuration...
warning: Git tree '/var/src/machine-config' is dirty
building '/nix/store/ddifxrf1yhmm7q42pbajypppdlj5ajns-homeassistant-2021.1.3.drv'...
error: --- Error ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix
builder for '/nix/store/ddifxrf1yhmm7q42pbajypppdlj5ajns-homeassistant-2021.1.3.drv' failed with exit code 1; last 10 log lines:
      app[KEY_BANNED_IPS] = await async_load_ip_bans_config(

  tests/components/shell_command/test_init.py::test_do_no_run_forever[pyloop]
    /nix/store/yi9a1fafijn0nf5m8qkmnl955vi2mncm-python3-3.8.6/lib/python3.8/asyncio/unix_events.py:917: RuntimeWarning: A loop is being detached from a child watcher with pending handlers
      warnings.warn(

  -- Docs: https://docs.pytest.org/en/stable/warnings.html
  =========================== short test summary info ============================
  ERROR tests/components/ssdp/test_init.py::test_scan_match_st[pyloop] - Runtim...
  ====== 2501 passed, 1 skipped, 11 warnings, 1 error in 169.19s (0:02:49) =======
error: --- Error ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix
1 dependencies of derivation '/nix/store/fkvq3xckqkhvb2c9p5rphb93a26ychdj-unit-home-assistant.service.drv' failed to build
error: --- Error ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix
1 dependencies of derivation '/nix/store/ablfc6156hwnjj2k0i5qs9ifp3l3g269-system-units.drv' failed to build
error: --- Error ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix
1 dependencies of derivation '/nix/store/2rhzvz6wkd47rf0p91vcs9nk4m8vfxsn-etc.drv' failed to build
error: --- Error ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix
1 dependencies of derivation '/nix/store/xmq5n6yixdzr7yxji7fagsphqdjhlv72-nixos-system-birne-21.03.20210117.68398d2.drv' failed to build
```

I'll probably disable the tests, but this seems like something that should be fixed anyway?

mweinelt commented:
As you can see the test is flaky, I appears, vanishes, reappears. I still want to see the exact test failure so I can make up my mind whether or not I'll report it upstream or just blacklist it downstream.

pinpox commented:
@mweinelt I'm running it with nixos-rebuild now as you can see (no krops), still not sure how to get more output to help

mweinelt commented:
Sorry, I have no idea what curbs your log output.
